pam/integration-tests/gdm-integration_test.go

```diff
@@ -0,0 +1,471 @@
+package main_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/msteinert/pam/v2"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/internal/testutils"
+	grouptests "github.com/ubuntu/authd/internal/users/localgroups/tests"
+	"github.com/ubuntu/authd/pam/internal/gdm"
+	"github.com/ubuntu/authd/pam/internal/gdm_test"
+	"github.com/ubuntu/authd/pam/internal/pam_test"
+	"github.com/ubuntu/authd/pam/internal/proto"
+)
+
+func init() {
+	gdm.AdvertisePamExtensions([]string{gdm.PamExtensionCustomJSON})
+}
+
+const (
+	exampleBrokerName = "ExampleBroker"
+	localBrokerName   = "local"
+	ignoredBrokerName = "<ignored-broker>"
+
+	passwordAuthID = "password"
+	fido1AuthID    = "fidodevice1"
+	phoneAck1ID    = "phoneack1"
+)
+
+func testGdmModule(t *testing.T, libPath string, args []string) {
+	t.Helper()
+
+	if !pam.CheckPamHasStartConfdir() {
+		t.Skip("can't test with this libpam version!")
```

Comment:
Unsure to understand how that can happen? Isn’t that rather a test setup failure or that we are testing on a version of a distro too old, and so we can’t run our tests?

I don’t think we should skip, because it means then that we didn’t try running the tests and can’t assert that it’s fine to ship on it. Does it make sense?

Reply:
Yeah, I suppose all the distros we care about nowadays support it, but I put it just in case

Reply:
let’s make it an error